### PR TITLE
[onert] Implement BinaryArithmeticLayer in train backend

### PR DESCRIPTION
**Pull request by Aeren1564**
This commit implements BinaryArithmeticLayer in train backend.

From draft https://github.com/Samsung/ONE/pull/12417

### CODE REVIEW

File: runtime/onert/backend/train/ops/BinaryArithmeticLayer.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                 const ArithmeticType arithmetic_type);
+  void forward(bool training) override;
+  void backward() override;
+
+private:
+  IPortableTensor *_back_prop_lhs;
+  IPortableTensor *_back_prop_rhs;
+  const IPortableTensor *_back_prop_output;
+
+  ArithmeticType _arithmetic_type;
```
Comment (zetwhite) [inline runtime/onert/backend/train/ops/BinaryArithmeticLayer.h:62]: (note) 
`cpu::ops::BinaryArithmeticLayer` doesn't hold activation. 
That's why this field is added.

File: runtime/onert/backend/train/ops/BinaryArithmeticLayer.cc [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  {
+    backprop_act =
+      backpropActivation(_activation, _output, _back_prop_output, _act_back_prop_output.get());
+  }
+  catch (const std::exception &e)
+  {
+    throw std::runtime_error{"train BinaryArithmeticLayer: " + std::string(e.what())};
+  }
+  assert(backprop_act != nullptr);
+
```
Comment (jyoungyun) [inline runtime/onert/backend/train/ops/BinaryArithmeticLayer.cc:76]: Please check the data type. If only the float data type is supported, the other type must throw an error.

File: runtime/onert/backend/train/ops/BinaryArithmeticLayer.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  ir::Activation _activation;
```
Comment (zetwhite) [inline runtime/onert/backend/train/ops/BinaryArithmeticLayer.h:63]: (note) 
`cpu::ops::BinaryArithmeticLayer` doesn't hold activation. 
That's why this field is added.

**Review by zetwhite — DISMISSED** [review]
Except this comment https://github.com/Samsung/ONE/pull/12542/files#r1469144238, 
Otherwise looks good :+1: :

File: runtime/onert/backend/train/ops/BinaryArithmeticLayer.cc (hunk 2 above)
Comment (Aeren1564) [reply runtime/onert/backend/train/ops/BinaryArithmeticLayer.cc:79]: Following check has been added.

```cpp
  if (_back_prop_output->data_type() != OperandType::FLOAT32)
    throw std::runtime_error{"Unsupported Data Type"};
```

**Review by jyoungyun — APPROVED** [review]
LGTM

**Review by zetwhite — APPROVED** [review]

**Pull request merged**